spec/requests/api/v1/targets/index_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe 'GET /api/v1/Targets', type: :request do
+  subject { get api_v1_targets_url, headers: headers, as: :json }
+
+  let!(:user) { create(:user) }
+  let(:topic) { create(:topic) }
+
+  context 'with logged-in user' do
+    let(:headers) { auth_headers }
+
+    context 'when the request is valid and the user has targets' do
+      let!(:targets) { create_list(:target, 5, user: user, topic: topic) }
```

Comment:
How can you be sure you are returning only the current user target? And not all targets? I would create some targets that does not belongs to the current user to check that.

Reply:
I did not realize to check it, will test now thank you.